statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/httpfn/HttpFunctionProvider.java

```diff
@@ -41,12 +48,56 @@ public RequestReplyFunction functionOfType(FunctionType type) {
     if (spec == null) {
       throw new IllegalArgumentException("Unsupported type " + type);
     }
+    return new RequestReplyFunction(
+        spec.states(), spec.maxNumBatchRequests(), buildHttpClient(spec));
+  }
+
+  private RequestReplyClient buildHttpClient(HttpFunctionSpec spec) {
+    // We need to build a UDS HTTP client
+    if (spec.isUnixDomainSocket()) {
+
+      // We need to split the path in order to get the sock file and the path after the sock file
+      Map.Entry<String, String> splittedFilePathAndEndpoint =
+          splitFilePathAndEndpointForUDS(spec.endpoint());
+
+      OkHttpClient specificClient =
+          sharedClient
+              .newBuilder()
+              .socketFactory(
+                  new AFUNIXSocketFactory.FactoryArg(splittedFilePathAndEndpoint.getKey()))
+              .callTimeout(spec.maxRequestDuration())
+              .build();
+
+      return new HttpRequestReplyClient(
+          // Only the path matters!
+          HttpUrl.get(URI.create(splittedFilePathAndEndpoint.getValue())), specificClient);
```

Comment:
There are two unfortunate things going on here:
1) `HttpUrl.get` would silently return `null` if it was unable to parse the URI
2) the endpoint only (rightfully so) contains the path without the schema and `HttpUrl` doesn't consider it to be valid.

Reply:
Ah, that's interesting... So I suppose we need a placeholder value like https://github.com/square/okhttp/blob/master/samples/unixdomainsockets/src/main/java/okhttp3/unixdomainsockets/ClientAndServer.java#L48

Reply:
Ok I changed the invocation of `HttpUrl.get` to use the `String` override, where the documentation states

```
@throws IllegalArgumentException If {@code url} is not a well-formed HTTP or HTTPS URL.
```

Hopefully this should avoid silent errors